### DISCUSSION TITLE
Direct Rules shortcut?

### DISCUSSION THREAD

**Discussion by archetyped** [objective-see/LuLu#577] [Q&A]
Hi, as I am frequently adding/updating rules in LuLu, I am curious whether there is a way to open the Rules window directly (or create a shortcut)?

Currently, accessing the Rules window requires first opening LuLu's settings or expanding LuLu's menu bar icon, neither of which can be done via the keyboard alone.

An option to set a global shortcut would be nice, but a command to directly open the Rules window would be even better as the window could be directly opened via Spotlight/Alfred/etc. as well.

**Comment by archetyped** [comment]
@objective-see any guidance on this?  Would this be better-suited as a support ticket?

**Reply by mdjunior** [reply]
I think that in the new versions being tested this is already possible. If you can test version 2.6.2, it already has the possibility of including a rule 1-click from the menu.

**Reply by archetyped** [reply]
I am currently using v2.6.2, but do not see any change that allows opening the Rules window directly.

**Comment by mdjunior** [comment]
Hi! In version 2.6.2, when using the menu it is possible to directly open the rules addition screen. Is this not working for you?

![Screenshot 2024-02-15 at 23 35 44](https://github.com/objective-see/LuLu/assets/3290669/c5f433c8-9394-44d9-9870-6649b0d44a08)

**Comment by objective-see** [comment]
The main issue here is that when LuLu is not active, any (app-specific) keyboard shortcuts are not relevant. So if we added say "cmd+r" to open the rules window, you'd still have to open LuLu's status menu first. As @mdjunior noted, I've added a (sub)menu option ...though I realize this still takes a few clicks. 

Open to other ideas, but not sure how to say do this via Spotlight, etc...

**Reply by archetyped** [reply]
A global (rather than app-specific) keyboard shortcut would be one option. 

However, something like a CLI flag (e.g. `LuLu --rules`) would provide a more flexible way to open the rules dialog using whatever means users choose (Terminal, Alfred integration, Automator, Shortcuts app, etc.).

**Comment by archetyped** [comment]
An alternative option would be a [URL scheme](https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app) to allow access to Lulu's functionality.

Examples:

```
lulu://rules/show # Open rules dialog
lulu://rules/add # Add rule
lulu://settings # Open settings
lulu://enable # Enable filtering
lulu://disable # Disable filtering
lulu://toggle-filtering # toggle filtering state
```

The actions with potential security implications could be coupled with a persistent notification (though a nefarious script could easily trigger an action via Lulu's menu bar without requiring any confirmation).
